### PR TITLE
Fix StrictMode in React 18

### LINKED ISSUE #254: This recaptcha instance has been already rendered
Hi @sarneeh 
I really like this library that does solve problems existent in other libraries, especially regarding multiple instances and react-router. 
This bug only exists in React >18, I have tried React 17 which works well.
Repeat: https://codesandbox.io/s/jovial-cookies-uclvr5?file=/src/App.js
Thanks for your attention.
WD

### PR DESCRIPTION
**Pull request by jgoz**
Fixes #254.

This PR moves the `timer` property from state to a component instance field, which fixes StrictMode usage in React 18.

React 18's StrictMode mounts, unmounts, and remounts components before actually rendering them in order to flush out any side effect cleanup issues. This presented as an error in Reaptcha because the component was being unmounted before `this.state.timer` was actually being set, which caused two copies of the interval to become active.

It turns out that `timer` doesn't actually need to be stored in state since it doesn't affect rendering. In other words, moving it to an instance field is arguably more "correct" and it ensures compatibility with StrictMode.

Note: in order to test this locally, I updated the react version in the examples to 18, which apparently affected the version used by the library's unit tests. I'd be happy to revert those changes and only leave the StrictMode fixes if you would prefer.

### CODE REVIEW

**Comment by jgoz** [comment]
@sarneeh Friendly ping - any thoughts on the approach I took here?

**Comment by sarneeh** [comment]
@jgoz Thank you very much for you contribution 🙏 I'm sorry for the late reply, I didn't have too much time to take a look at this.

> Note: in order to test this locally, I updated the react version in the examples to 18, which apparently affected the version used by the library's unit tests. I'd be happy to revert those changes and only leave the StrictMode fixes if you would prefer.

I have to move the tests to something different (probably RTL) anyways as Enzyme seems dead. For now using the unofficial adapter is fine.

File: lib/index.tsx [hunk 1]
```diff
@@ -78,6 +77,7 @@ const PROPS_THAT_SHOULD_CAUSE_RERENDER: Array<keyof RecaptchaBaseConfig> = [
 
 class Reaptcha extends Component<Props, State> {
   container?: HTMLDivElement | null;
+  timer?: number | undefined;
```
Comment (sarneeh) [inline lib/index.tsx:80]: When the property is optional (with `?`), it's automatically `type | undefined`, so the explicit `| undefined` is not needed. But it's just cosmetics.

**Review by sarneeh — APPROVED** [review]
LGTM

**Comment by jgoz** [comment]
Thank you for looking! Much appreciated.

**Comment by rafakwolf** [comment]
Still facing the same error on NextJS 12
Reaptcha version : "reaptcha": "^1.12.1"
image

**Comment by sarneeh** [comment]
@rafakwolf Would need some kind of reproduction to understand what's going on 👀

**Comment by rafakwolf** [comment]
@sarneeh https://github.com/rafakwolf/nextjs-captcha

**Comment by sarneeh** [comment]
@rafakwolf Are the screenshots above from the reproduction repo? I just tested it and [it works on my side](https://jsarnowski-dropshare.s3-eu-central-1.amazonaws.com/Kapture-2022-07-27-at-07.22.44.mp4) 🤔

**Pull request merged**